### PR TITLE
Allow specifying multiple Jenkins URLs

### PR DESCRIPTION
**Pull request by nre-ableton**
Ping @AbletonDevTools/gotham-city

### CODE REVIEW

**Review by rco-ableton — APPROVED** [review]
LGTM 👍

File: Jenkinsfile [hunk 1]
```diff
@@ -49,7 +49,7 @@ devToolsProject.run(
       '8000',
       '-v jenkins-nodes:/jenkins_nodes',
       '--target-port 9100 --target-port 9323' +
-        " ${env.JENKINS_URL} /jenkins_nodes/output.json",
+        " --url ${env.JENKINS_URL} /jenkins_nodes/output.json",
```
Comment (ala-ableton) [inline Jenkinsfile:52]: This change belongs to _Move Jenkins URL to required --url argument_ instead of to _Allow specifying multiple Jenkins URLs_.

**Review by ala-ableton — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (nre-ableton) [reply Jenkinsfile:52]: Right you are!

File: jenkins_node_scanner.py [hunk 2]
```diff
@@ -224,6 +216,14 @@ def main():
     while True:
         with ignore_exceptions_except_exit():
             with GLOBAL_EXCEPTIONS.count_exceptions():
+                logging.info('Connecting to Jenkins master at %s', args.url)
```
Comment (ala-ableton) [inline jenkins_node_scanner.py]: How much spam is this producing then?

Comment (nre-ableton) [reply jenkins_node_scanner.py]: Not much, but I guess it should be a debug line.

File: jenkins_node_scanner.py [hunk 3]
```diff
@@ -224,30 +243,22 @@ def main():
...
+                for url in args.urls:
+                    logging.debug('Connecting to Jenkins master at %s', url)
+                    master = jenkins.Jenkins(
+                        url,
+                        username=args.username,
+                        password=args.password,
+                        timeout=args.timeout,
+                    )
+
+                    node_infos.append(
```
Comment (mst-ableton) [inline jenkins_node_scanner.py]: Shouldn't this be `node_infos.extend(...)`?

**Review by mst-ableton — CHANGES_REQUESTED** [review]
lgtm except for one change

File: jenkins_node_scanner.py (hunk 3 above)
Comment (nre-ableton) [reply jenkins_node_scanner.py]: Good catch, fixed in 475a696f6cdb387c69ad2323e1c947dcb85464f9

File: jenkins_node_scanner.py [hunk 4]
```diff
@@ -158,6 +158,27 @@ def get_node_ip(master, node):
     return description.split('from ')[1].split(' :')[0]
 
 
+def get_node_info(master, target_ports, exclude_regex):
```
Comment (ala-ableton) [inline jenkins_node_scanner.py]: This function return several items in a list, so it should be named `get_node_infos` or `get_nodes_info`.

**Review by ala-ableton — CHANGES_REQUESTED** [review]

File: jenkins_node_scanner.py (hunk 3 above)
Comment (ala-ableton) [reply jenkins_node_scanner.py]: Then I guess `get_node_info` could have a better name.

File: jenkins_node_scanner.py (hunk 4 above)
Comment (nre-ableton) [reply jenkins_node_scanner.py]: Agreed. I prefer the first option. Fixed in 6ebf4502ce8450894f658b07efd5d04cbbe8d1fa.

**Comment by ala-ableton** [comment]
@nre-ableton please rebase/squash and I'll do another review, thanks!

**Comment by nre-ableton** [comment]
@ala-ableton done!

File: jenkins_node_scanner.py (hunk 2 above)
Comment (ala-ableton) [inline jenkins_node_scanner.py]: I think the `logging.info` -> `logging.debug` change belongs to _Move Jenkins object declaration closer to usage_.

File: jenkins_node_scanner.py [hunk 5]
```diff
@@ -224,28 +245,7 @@ def main():
...
-
-                node_info = [{
-                    'labels': {
-                        'jenkins_master': args.url,
-                        'node': node['name'],
-                    },
-                    'targets': ['%s:%d' % (ip_addr, port) for port in args.target_ports
-                                if is_port_open(ip_addr, port)],
-                } for node, ip_addr in node_ips]
+                node_info = get_node_infos(master, args.target_ports, exclude_regex)
```
Comment (ala-ableton) [inline jenkins_node_scanner.py]: Please rename `node_info` to `node_infos` to match `get_node_infos`.

**Review by ala-ableton — CHANGES_REQUESTED** [review]

**Review by ala-ableton — APPROVED** [review]
Very nice!
LGTM :+1:

**Pull request merged**